libtransmission/announcer.cc

```diff
@@ -989,180 +968,183 @@ static void onAnnounceDone(tr_announce_response const* response, void* vdata)
     time_t const now = tr_time();
     tr_announce_event const event = data->event;
 
-    if (tier != nullptr)
+    if (tier == nullptr)
     {
-        tr_logAddTraceTier(
-            tier,
-            fmt::format(
-                "Got announce response: "
-                "connected:{} "
-                "timeout:{} "
-                "seeders:{} "
-                "leechers:{} "
-                "downloads:{} "
-                "interval:{} "
-                "min_interval:{} "
-                "tracker_id_str:{} "
-                "pex:{} "
-                "pex6:{} "
-                "err:{} "
-                "warn:{}",
-                response->did_connect,
-                response->did_timeout,
-                response->seeders,
-                response->leechers,
-                response->downloads,
-                response->interval,
-                response->min_interval,
-                (!std::empty(response->tracker_id) ? response->tracker_id.c_str() : "none"),
-                std::size(response->pex),
-                std::size(response->pex6),
-                (!std::empty(response->errmsg) ? response->errmsg.c_str() : "none"),
-                (!std::empty(response->warning) ? response->warning.c_str() : "none")));
-
-        tier->lastAnnounceTime = now;
-        tier->lastAnnounceTimedOut = response->did_timeout;
-        tier->lastAnnounceSucceeded = false;
-        tier->isAnnouncing = false;
-        tier->manualAnnounceAllowedAt = now + tier->announceMinIntervalSec;
-
-        if (response->external_ip)
-        {
-            data->session->setExternalIP(*response->external_ip);
-        }
+        delete data;
+        return;
+    }
```

Comment:
Minor optimization: we could move this `if ()` before the declaration of `now` and `event`.

Reply:
moved in 79b51a742cb7d454cd6e43b7589b093bdb88c2ba